apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -219,15 +223,13 @@ def concat(self) -> _RT:
         """
         raise NotImplementedError("Blockwise iterators do not support concat operation")
 
-    _EagerRT = TypeVar("_EagerRT")
-
     def _maybe_eager_iterator(
         self, x: Iterator[_EagerRT], _pool: Optional[ThreadPoolExecutor] = None
     ) -> Iterator[_EagerRT]:
         """Private"""
         return EagerIterator(x, pool=_pool) if self.eager else x
 
-    def _table_reader(self) -> BlockwiseSingleAxisTableIter:
+    def _table_reader(self) -> Generator[BlockwiseTableReadIterResult, None, None]:
```

Comment:
Since this generator only does the iteration part, [`Iterator[BlockwiseTableReadIterResult]`](https://docs.python.org/3.9/library/typing.html#typing.Generator:~:text=Alternatively%2C%20annotate%20your%20generator%20as%20having%20a%20return%20type%20of%20either%20Iterable%5BYieldType%5D%20or%20Iterator%5BYieldType%5D%3A) might communicate the intent better (and avoid making promises we don’t intend to keep).